src/v/security/audit/audit_log_manager.cc

```diff
@@ -847,22 +847,20 @@ audit_log_manager::should_enqueue_audit_event(
         return std::make_optional(audit_event_passthrough::yes);
     }
 
-    if (auto val = should_enqueue_audit_event(); val.has_value()) {
-        return val;
-    }
     if (!is_audit_event_enabled(kafka_api_to_event_type(api))) {
         return std::make_optional(audit_event_passthrough::yes);
     }
 
+    if (auto val = should_enqueue_audit_event(); val.has_value()) {
```

Comment:
No wonder there are bugs!

How about we replace all of these with:
```c++
std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(event_type type) const {
    if (!is_audit_event_enabled(type)) {
        return std::make_optional(audit_event_passthrough::yes);
    }
    return should_enqueue_audit_event();
}

std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(
  event_type type, const security::acl_principal& principal) const {
    if (_audit_excluded_principals.contains(principal)) {
        return std::make_optional(audit_event_passthrough::yes);
    }
    return should_enqueue_audit_event(type);
}

std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(
  kafka::api_key key,
  const security::acl_principal& principal,
  const model::topic& t) const {
    if (_audit_excluded_topics.contains(t)) {
        return std::make_optional(audit_event_passthrough::yes);
    }
    return should_enqueue_audit_event(key, principal);
}

std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(
  kafka::api_key api, const security::acl_principal& principal) const {
    return should_enqueue_audit_event(kafka_api_to_event_type(api), principal);
}

std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(
  event_type type, const security::audit::user& user) const {
    return should_enqueue_audit_event(
      type, security::acl_principal{security::principal_type::user, user.name});
}

std::optional<audit_log_manager::audit_event_passthrough>
audit_log_manager::should_enqueue_audit_event(
  event_type type, const ss::sstring& username) const {
    return should_enqueue_audit_event(
      type, security::acl_principal{security::principal_type::user, username});
}
```
I didn't see any changes to tests, so maybe some tests need adding?